fluent-bundle/src/types.rs

```diff
@@ -111,14 +111,60 @@ impl<'source> From<&ast::InlineExpression<'source>> for DisplayableNode<'source>
     }
 }
 
-#[derive(Debug, PartialEq, Clone)]
+pub trait FluentType: std::fmt::Debug {
+    fn duplicate(&self) -> Box<dyn FluentType>;
+    fn get_hash(&self) -> (&'static str, usize);
+    fn to_string(&self) -> Cow<'static, str>;
+    fn display(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result;
```

Comment:
Why not just add a Display bound?

Reply:
I had to rename `to_string` to `as_string` in result of that to not override the Display's method, but I think that it works well.